### DISCUSSION TITLE
I really need help with Templates? How to interpret and use them?

### DISCUSSION THREAD

**Discussion by viperwasp** [oobabooga/text-generation-webui#4469] [Q&A]
Here is a question I asked. 
https://huggingface.co/TheBloke/openchat_3.5-GGUF/discussions/5

I am also asking it here to see where I can get help the fastest etc? 

Here is my question. 
For the model [TheBloke](https://huggingface.co/TheBloke) /[openchat_3.5-GGUF](https://huggingface.co/TheBloke/openchat_3.5-GGUF)

In the readme it says...

prompt_template: |
GPT4 User: {prompt}<|end_of_turn|>GPT4 Assistant:

How do I use this in instruct mode in Oobabooga? Or do I have to use chat mode? I prefer having a long discussion with the AI not role play but asking questions and getting it to write stories based on what we discussed etc. Kind of like how one would use ChatGPT. Basically how can I add that template into Ooba? I've been to the Instruct template tab. I know how to change templates. But not how to properly format or add new ones etc.

**Comment by HiroseKoichi** [comment]
I know it's a bit late and you've likely already figured it out, but here is a link to the wiki entries:
[How to create an instruction template.](https://github.com/oobabooga/text-generation-webui/wiki/03-%E2%80%90-Parameters-Tab#instruction-template)
[What the difference between chat, chat-instruct, and instruct is.](https://github.com/oobabooga/text-generation-webui/wiki/01-%E2%80%90-Chat-Tab#mode)

**Reply by viperwasp** [reply]
Nope I have not figured it out and someone else asked me how to do it recently both of us were lost. So I will read this over and thank you.

**Reply by Kvisaz** [reply]
Hi there are simple definitions, but we need explanations and samples

As i see, "prompt templates" in official models pages and "prompt templates" in Oobabooga are completely different.

Many good models, especially Llama 3 or Nvidia Chat QA, are totally unusable in Oobabooga format.

So I try to migrate to another package with more transparent documentation and leave oobabooga
